client/src/stores/DataStore.ts

```diff
@@ -13,8 +13,21 @@ class DataStore {
 
     const res = await axios.get("/api/subjects");
     this.subjectsLoading = false;
-    this.subjects = res.data;
+    console.log(res.data.error);
+    if (!res.data.error) {
+      this.subjects = res.data;
+    }
+  }
+
+  async getSubjectById(subjectId: string) {
+    this.subjectsLoading = true;
 
+    const res = await axios.get(`/api/subjects/${subjectId}`);
+    this.subjectsLoading = false;
+    console.log("dataaaaaa: ",res.data);
+    if (res.data) {
+      return res.data;
```

Comment:
You are returning here data and letting the UI Component manage the state.
We should try managing the state/logic seperated from the UI.
You can use here a Map to manage requests for diffrent ids.
Talk to me we can go through this live.

Reply:
would love for another explanation